RecoTracker/FinalTrackSelectors/plugins/TrackSelectorByRegion.cc

```diff
@@ -0,0 +1,166 @@
+          phiMax.reserve(n_regions);
+          std::vector<float> phi0;
+          phi0.reserve(n_regions);
+          std::vector<float> phi0margin;
+          phi0margin.reserve(n_regions);
+          std::vector<math::XYZPoint> origin;
+          origin.reserve(n_regions);
+          std::vector<float> zBound;
+          zBound.reserve(n_regions);
+
+          for (const auto& tmp : regions)
+            if (const auto* etaPhiRegion = dynamic_cast<const RectangularEtaPhiTrackingRegion*>(&tmp)) {
+              const auto& etaRange = etaPhiRegion->etaRange();
+              const auto& phiMargin = etaPhiRegion->phiMargin();
+
+              etaMin.push_back(etaRange.min());
+              etaMax.push_back(etaRange.max());
+
+              phiMin.push_back(etaPhiRegion->phiDirection() - phiMargin.left());
+              phiMax.push_back(etaPhiRegion->phiDirection() + phiMargin.right());
+              phi0.push_back(etaPhiRegion->phiDirection());
+              phi0margin.push_back(phiMargin.right());
+
+              GlobalPoint gp = etaPhiRegion->origin();
+              origin.push_back(math::XYZPoint(gp.x(), gp.y(), gp.z()));
+              zBound.push_back(etaPhiRegion->originZBound());
+            }
+
+          size_t it = 0;
+          //	    std::cout << "tracks: " << tracks.size() << std::endl;
+          for (auto trk : tracks) {
+            const auto pt = trk.pt();
+            if (pt < minPt_) {
+              //		std::cout << " KO !!! for pt" << std::endl;
+              continue;
+            }
+
+            const auto eta = trk.eta();
+            const auto phi = trk.phi();
+
+            for (size_t k = 0; k < n_regions; k++) {
+              //		if ( std::abs(trk.vz() - origin[k].z()) > zBound[k] ) {
+              //		std::cout << "std::abs(trk.vz() - origin[k].z()): " << std::abs(trk.vz() - origin[k].z()) << " zBound: " << zBound[k] << std::endl;
+              //		std::cout << "std::abs(trk.dz(origin[k])): " << std::abs(trk.dz(origin[k])) << " zBound: " << zBound[k] << std::endl;
+              if (std::abs(trk.dz(origin[k])) > zBound[k]) {
+                //		  std::cout << " KO !! for z" << std::endl;
+                continue;
+              }
+              //		std::cout << "eta : " << eta << " etaMin[k]: " << etaMin[k] << " etaMax[k]: " << etaMax[k] << std::endl;
+              if (eta < etaMin[k]) {
+                //		  std::cout << " KO !!! for eta" << std::endl;
+                continue;
+              }
+              if (eta > etaMax[k]) {
+                //		  std::cout << " KO !!! for eta" << std::endl;
+                continue;
+              }
+              //		std::cout << "phi: " << phi << " phi0[k]: " << phi0[k] << " std::abs(reco::deltaPhi(phi,phi0[k])): " << std::abs(reco::deltaPhi(phi,phi0[k])) << " phi0margin: " << phi0margin[k] << std::endl;
+              if (std::abs(reco::deltaPhi(phi, phi0[k])) > phi0margin[k] * 1.1) {
+                //		  std::cout << " KO !!! for phi" << std::endl;
```

Comment:
is this commented out code needed?
If debugging is still expected, please add it with LogTrace or LogDebug or similar so that necessary parts can be checked without rewriting the code